source/server/http/admin.cc

```diff
@@ -275,6 +276,36 @@ Http::Code AdminImpl::handlerClusters(const std::string&, Http::HeaderMap&,
   return Http::Code::OK;
 }
 
+// TODO(jsedgwick) Use query params to list available dumps, selectively dump, etc
+Http::Code AdminImpl::handlerConfigDump(const std::string&, Http::HeaderMap&,
+                                        Buffer::Instance& response) const {
+  envoy::admin::ConfigDump dump;
+  auto& config_dump_map = *(dump.mutable_configs());
+  for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
+    ProtobufTypes::MessagePtr message;
+    try {
+      message = key_callback_pair.second();
+    } catch (const std::exception& e) {
```

Comment:
By crash, do you mean RELEASE_ASSERTING or letting the exception escape? Presumably the latter will also crash us, just clarifying your request.

I think you mentioned the desire to be able to *dynamically* hook in plugins and so on, so I was coding defensively. Is the stance going to be that we won't try to recover from any exception thrown by someone's plugin? It seems that segfaults and such aside, we don't want the rest of envoy to go down when you curl an admin endpoint (i.e. at some unpredictable time after start) because of something outside of this repo's control. So swallowing/logging would be fine here from that point of view because for this code the exception is recoverable.

Just asking about the philosophy here, can definitely change this and just crash.

Reply:
Per above take a look at the philosophy document. Our general philosophy is fail hard when possible, make it nicer later if needed. In this case I would just let the exceptions propagate which will crash the server hard (for now).